File: game_client/Assets/Hotfix~/Hotfix/Assets/Protocol/Common/NumMergeObj/NumMerge_BlockList.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using ALBasicProtocolPack;


namespace Hotfix.Common.NumMergeObj
{

/// <summary>
/// 数字合并-棋子列表
/// </summary>
public class NumMerge_BlockList : ALBasicProtocolPack._IALProtocolStructure {
/// <summary>
/// 棋子列表
/// </summary>
private List<Hotfix.Common.NumMergeObj.NumMerge_BlockBase> blocks;


public NumMerge_BlockList() {
	blocks = new List<Hotfix.Common.NumMergeObj.NumMerge_BlockBase>();
}

public NumMerge_BlockList(
	List<Hotfix.Common.NumMergeObj.NumMerge_BlockBase> _blocks
) {	blocks = _blocks;
}

public byte getMainOrder() { return (byte)0; }

public byte getSubOrder() { return (byte)0; }

/// <summary>
/// 棋子列表
/// </summary>
public List<Hotfix.Common.NumMergeObj.NumMerge_BlockBase> getBlocks() { return blocks; }
/// <summary>
/// 棋子列表
/// </summary>
public void addBlocks(Hotfix.Common.NumMergeObj.NumMerge_BlockBase _blocks) { blocks.Add(_blocks); }


public int GetBufSize() {
	int _size = 0;
	_size += 2 + (blocks.Count * 12);

	return _size;
}

public int GetFullPackBufSize() {
	int _size = 2;
	_size += 2 + (blocks.Count * 12);

	return _size;
}



public void ReadUnzipBuf(ALProtocolBuf _buf, int _finalPos) {
	 if(_finalPos > 0 && _buf.getCurPos() >= _finalPos) return ;
	short _blocksCount = _buf.getShort();
	for(int _i = 0; _i < _blocksCount; _i++) { 
		Hotfix.Common.NumMergeObj.NumMerge_BlockBase _blocks = new Hotfix.Common.NumMergeObj.NumMerge_BlockBase();
		int __blocksCustLen = _buf.getInt();
	int __blocksCurPos = _buf.getCurPos();
	_blocks.ReadUnzipBuf(_buf, __blocksCurPos + __blocksCustLen);
	_buf.setPosition(__blocksCurPos + __blocksCustLen);

		blocks.Add(_blocks);
	}
}

public void PutUnzipBuf(ALProtocolBuf _buf) {
	_buf.putShort((short)blocks.Count);
	for(int _i = 0; _i < blocks.Count; _i++) { 
		_buf.putInt(blocks[_i].GetBufSize());
	blocks[_i].PutUnzipBuf(_buf);
	}
}

public byte[] makeFullPackage() {
	int _bufSize = GetBufSize() + 2;
	ALProtocolBuf _buf = ALProtocolBuf.allocate(_bufSize);
	_buf.put((byte)0);
	_buf.put((byte)0);
	PutUnzipBuf(_buf);
	return _buf.getBuf();
}
public void makeFullPackage(ALProtocolBuf _recBuf) {
	if(null == _recBuf)
		return ;
	_recBuf.put((byte)0);
	_recBuf.put((byte)0);
	PutUnzipBuf(_recBuf);
}
public byte[] makePackage() {
	int _bufSize = GetBufSize();
	ALProtocolBuf _buf = ALProtocolBuf.allocate(_bufSize);
	PutUnzipBuf(_buf);
	return _buf.getBuf();
}
public void readPackage(byte[] _buf) {
	ALProtocolBuf _bufObj = new ALProtocolBuf(_buf);
	ReadUnzipBuf(_bufObj, -1);
}
public void readPackage(ALProtocolBuf _buf) {
	ReadUnzipBuf(_buf, -1);
}
public override string ToString() {
	System.Text.StringBuilder builder = new System.Text.StringBuilder();

	builder.Append("{");
	builder.Append("blocks").Append(":").Append(blocks.ToString()).Append(", ");
	builder.Append("}");
	return builder.ToString();
}

}

}

File: game_server/ActivitiesV02/src/Hotfix/V02/Common/NumMergeObj/NumMerge_BlockList.java
package Hotfix.V02.Common.NumMergeObj;

import java.nio.ByteBuffer;
/*********
 * 数字合并-棋子列表
 **/
public class NumMerge_BlockList implements ALBasicProtocolPack._IALProtocolStructure {
/** 棋子列表 */
private java.util.ArrayList<Hotfix.V02.Common.NumMergeObj.NumMerge_BlockBase> blocks;


public NumMerge_BlockList() {
	blocks = new java.util.ArrayList<Hotfix.V02.Common.NumMergeObj.NumMerge_BlockBase>();
}

public NumMerge_BlockList(
	 java.util.ArrayList<Hotfix.V02.Common.NumMergeObj.NumMerge_BlockBase> _blocks
) {	blocks = _blocks;
}

public final byte getMainOrder() { return (byte)0; }

public final byte getSubOrder() { return (byte)0; }

/** 棋子列表 */
public java.util.ArrayList<Hotfix.V02.Common.NumMergeObj.NumMerge_BlockBase> getBlocks() { return blocks; }
/** 棋子列表 */
public void addBlocks(Hotfix.V02.Common.NumMergeObj.NumMerge_BlockBase _blocks) { blocks.add(_blocks); }


public final int GetBufSize() {
	int _size = 0;
	_size += 2 + (blocks.size() * 12);

	return _size;
}

public final int GetFullPackBufSize() {
	int _size = 2;
	_size += 2 + (blocks.size() * 12);

	return _size;
}



public final void ReadUnzipBuf(ByteBuffer _buf, int _finalPos) {
	 if(_finalPos > 0 && _buf.position() >= _finalPos) return ;
	if(_buf.remaining() <= 0) return;
	short _blocksCount = _buf.getShort();
	for(int _i = 0; _i < _blocksCount; _i++) { 
		Hotfix.V02.Common.NumMergeObj.NumMerge_BlockBase _blocks = new Hotfix.V02.Common.NumMergeObj.NumMerge_BlockBase();
		if(_buf.remaining() <= 0) return;
	int __blocksCustLen = _buf.getInt();
	int __blocksCurPos = _buf.position();
	_blocks.ReadUnzipBuf(_buf, __blocksCurPos + __blocksCustLen);
	_buf.position(__blocksCurPos + __blocksCustLen);

		blocks.add(_blocks);
	}
}

public final void PutUnzipBuf(ByteBuffer _buf) {
	_buf.putShort((short)blocks.size());
	for(int _i = 0; _i < blocks.size(); _i++) { 
		_buf.putInt(blocks.get(_i).GetBufSize());
	blocks.get(_i).PutUnzipBuf(_buf);
	}
}

public final ByteBuffer makeFullPackage() {
	int _bufSize = GetBufSize() + 2;
	ByteBuffer _buf = ByteBuffer.allocate(_bufSize);
	_buf.put((byte)0);
	_buf.put((byte)0);
	PutUnzipBuf(_buf);
	_buf.flip();
	return _buf;
}
public final void makeFullPackage(ByteBuffer _recBuf) {
	if(null == _recBuf)
		return ;
	_recBuf.put((byte)0);
	_recBuf.put((byte)0);
	PutUnzipBuf(_recBuf);
}
public final ByteBuffer makePackage() {
	int _bufSize = GetBufSize();
	ByteBuffer _buf = ByteBuffer.allocate(_bufSize);
	PutUnzipBuf(_buf);
	_buf.flip();
	return _buf;
}
public final void readPackage(ByteBuffer _buf) {
	ReadUnzipBuf(_buf, -1);
}
}

